Source/Core/Core/PowerPC/BreakPoints.cpp

```diff
@@ -326,22 +324,19 @@ bool MemChecks::ToggleEnable(u32 address)
   return true;
 }
 
-bool MemChecks::Remove(u32 address, bool update)
+std::optional<DelayedMemCheckUpdate> MemChecks::Remove(u32 address)
 {
   const auto iter =
       std::find_if(m_mem_checks.cbegin(), m_mem_checks.cend(),
                    [address](const auto& check) { return check.start_address == address; });
 
   if (iter == m_mem_checks.cend())
-    return false;
+    return std::nullopt;
```

Comment:
Would it make sense to instead return a `DelayedMemCheckUpdate(nullptr, false)` here and eliminate the `optional` return-type and operator overload?